shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/Server.java

```diff
@@ -2649,7 +2649,7 @@ public void downloadInputData(
        * earlier. Do i need to release the inputDownloadSemaphore for any failures here too? I think
        * this increments the semaphore too much but I can't tell what the proper behaviour is
        */
-      inputDownloadSemaphore.release();
+      if (!readStale) inputDownloadSemaphore.release();
```

Comment:
Great! Would you mind removing the FIXME comment as well?

Reply:
fixed